_source/_posts/2022-01-30-local-https-java.adoc

```diff
@@ -0,0 +1,285 @@
+
+    @GetMapping("/")
+    public String sayHello() {
+        return "Yes, this is another Hello World...";
+    }
+}
+----
+
+That's it, start the application! Either from your IDE or by running `./mvnw spring-boot:run`.
+
+Now access this boring, non-secure application in your browser at`\http://localhost:8080/`.
+
+== HTTPS using an Internet proxy
+
+One of the quickest ways to get a local application running with TLS is to use an internet proxy like ngrok.
+https://ngrok.com/download[Download and install ngrok] and then start up the service using:
+
+[source,bash]
+----
+ngrok http 8080
+----
+
+This command will expose your local service running port `8080` over the internet on a _generated_ URL such as `\https://abcd-11-22-33-44.ngrok.io`.
+
+If you don't want to install ngrok, you could use https://localhost.run/[localhost.run], which works using SSH port forwarding; try it out with:
+
+[source,bash]
+----
+ssh -R 80:localhost:8080 nokey@localhost.run
+----
+
+Similar to ngrok, a temporary URL will be generated. For example, `\https://152ccb7ac9369e.lhr.life`.
+
+The goal of these tools is to provide a publicly available URL for your local development, which is perfect if you want to share something via the internet. If you don't, it's best not to expose your development machine to any potential attackers online. The following two options avoid this issue.
+
+== Create a certificate authority with mkcert
+
+The next two options use a tool called https://github.com/FiloSottile/mkcert[`mkcert`] which sets up a locally trusted certificate authority (CA), installed into the trust stores on your computer. Any certificates issued by this CA will be trusted by the client of your choice (Chrome, Firefox, curl, etc).
+
+On macOS, you can install `mkcert` with Homebrew; for other operating systems you can find instructions in the https://github.com/FiloSottile/mkcert#installation[`mkcert` docs].
+
+[source,bash]
+----
+brew install mkcert nss
+# nss is only needed if you are using Firefox
+----
+
+Create and install the certificate authority:
+
+====
+[source,bash]
+----
+mkcert -install
+----
+
+[source,txt]
+----
+Created a new local CA 💥
+Sudo password:
+The local CA is now installed in the system trust store! ⚡️
+The local CA is now installed in Java's trust store! ☕️
+----
+====
+
+[IMPORTANT]
+====
+
+If you are on macOS and using https://httpie.io/[HTTPie], you will need to add the CA to https://github.com/httpie/httpie/issues/480[HTTPie manually]: +
+
+[source,bash]
+----
+cat "$(mkcert --CAROOT)/rootCA.pem" >> \
+$(brew --prefix)/Cellar/httpie/*/libexec/lib/python*/site-packages/certifi/cacert.pem
+----
+====
+
+That was _mostly_ painless, now let's secure the application you just created!
+
+== Reverse proxy with Nginx
+
+Many production applications terminate TLS using a reverse proxy like Nginx; this can also be an excellent option for local development and has the added advantage of working with any web application framework.
```

Comment:
```suggestion
Many production applications terminate TLS using a reverse proxy like Nginx. This can also be an excellent option for local development and has the added advantage of working with any web application framework.
```
breaks up a long sentence